pkg/preparer/orchestrate.go

```diff
@@ -150,22 +151,28 @@ func (p *Preparer) handlePods(podChan <-chan pods.Manifest, quit <-chan struct{}
 			if !working {
 				p.tryRunHooks(hooks.AFTER_AUTH_FAIL, pods.NewPod(manifestToLaunch.ID(), pods.PodPath(p.podRoot, manifestToLaunch.ID())), &manifestToLaunch, manifestLogger)
 			}
-		case <-time.After(1 * time.Second):
+		default:
 			if working {
 				pod := pods.NewPod(manifestToLaunch.ID(), pods.PodPath(p.podRoot, manifestToLaunch.ID()))
 
 				// HACK ZONE. When we have better authz, rewrite.
-				// Still need to ensure that preparer launches correctly
-				// as root
-				if pod.Id == POD_ID {
-					pod.RunAs = "root"
-				}
-
-				ok := p.installAndLaunchPod(&manifestToLaunch, pod, manifestLogger)
-				if ok {
-					manifestToLaunch = pods.Manifest{}
+				if (manifestToLaunch.RunAsUser() == "root") && (pod.Id != POD_ID) {
```

Comment:
ah yeah you're right that it's not equivalent. Taking the `sleep()` out of the `else` would fix that.  And yeah in its old form my test would always take a second or more because this function always waits at least a second to do work. If I do the refactor I can put it back the way it was.

Although I think the intention of the code is clearer if we use a sleep?